views/index.py

```diff
@@ -61,97 +54,86 @@ def index():
+                    'mail': refine_data(user_ldap, 'mail'),
+                    'givenName': refine_data(user_ldap, 'givenName'),
+                    'sn': refine_data(user_ldap, 'sn'),
+                    'employeeID': refine_data(user_ldap, 'employeeID')
+                }
+                user_db = User.query.filter_by(
+                    employee_id=user_ldap_data['employeeID']
+                ).first()
+                if not user_db:
                     new_user = User(
-                        email=form.email.data,
-                        first_name=form.first_name.data,
-                        surname=form.surname.data,
-                        password_hash=generate_password_hash(
-                            form.password.data))
-                    send_confirmation_email(new_user.email)
+                        email=user_ldap_data['mail'],
+                        first_name=user_ldap_data['givenName'],
+                        surname=user_ldap_data['sn'],
+                        employee_id=user_ldap_data['employeeID'],
+                        active=True
+                    )
                     db.session.add(new_user)
                     db.session.commit()
-            except (ValueError, TypeError):
-                message_body = 'Registration failed'
-                message_title = 'Error!'
-                return render_template('message.html',
-                                       message_title=message_title,
-                                       message_body=message_body)
-        else:
-            return render_template('registration.html',
-                                   form=form,
-                                   error=form.errors)
-        message_body = 'The registration was successful.'
-        message_title = 'Success!'
-        return render_template('message.html',
-                               message_title=message_title,
-                               message_body=message_body)
+                else:
+                    user_db_data = {
+                        'mail': user_db.email,
+                        'givenName': user_db.first_name,
+                        'sn': user_db.surname,
+                        'employeeID': user_db.employee_id
+                    }
+                    if user_db_data != user_ldap_data:
+                        user_db.email = user_ldap_data['mail']
+                        user_db.first_name = user_ldap_data['givenName']
+                        user_db.surname = user_ldap_data['sn']
+                        user_db.employee_id = user_ldap_data['employeeID']
+                        db.session.commit()
+
+                user_db = User.query.filter_by(
+                    employee_id=user_ldap_data['employeeID']
+                ).first()
+                session['logged_in'] = True
+                session['id'] = user_db.id
+                session['email'] = user_db.email
+                print(session)
```

Comment:
I do not think it's secure to print user's session data to server logs. :)

Reply:
It is debugging artifact, deleted